src/main/java/org/spongepowered/common/config/DuplicateRemovalVisitor.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.config;
+
+import com.google.common.base.Preconditions;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.ConfigurationVisitor;
+import ninja.leaping.configurate.Types;
+
+import java.util.Objects;
+
+/**
+ * Given a configuration node, and a node at the same position in the tree of a parent configuration, remove values from the child node that don't
+ * change anything from the parent node.
+ */
+class DuplicateRemovalVisitor implements ConfigurationVisitor.Safe<ConfigurationNode, Void> {
```

Comment:
So what's this for exactly?

Reply:
This is the logic that removes values from, say a `dimension.conf` that are the same as values in `global.conf`, but migrated from `ConfigurationNodeWalker` to `ConfigurationVisitor`

Reply:
Just might want to include that as a more specific case of how it's being used in the class java doc.